File: Modules/Common/Equipments/SIE.Web.Fixtures/Repairs/Commands/AddFixtureRepairCommand.cs
using SIE.Fixtures;
using SIE.Fixtures.Repairs.ViewModels;
using SIE.Web.Command;
using System;

namespace SIE.Web.Fixtures.Repairs.Commands
{
    /// <summary>
    /// 添加工治具报修命令
    /// </summary>
    [JsCommand("SIE.Web.Fixtures.Repairs.Commands.AddFixtureRepairCommand")]
    public class AddFixtureRepairCommand : ViewCommand
    {
        /// <summary>
        /// 执行工治具报修命令
        /// </summary>
        /// <param name="args">args</param>
        /// <param name="scope">scope</param>
        /// <returns>执行结果</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            AddFixtureRepairInfo repairInfo = new AddFixtureRepairInfo();
            repairInfo.errMsg = string.Empty;
            try
            {
                repairInfo = RT.Service.Resolve<CoreFixtureController>().GetFixtureRepairInfo();
            }
            catch (Exception ex)
            {
                repairInfo.errMsg = ex.Message;
            }

            return repairInfo;
        }
    }
}

File: Modules/Common/Equipments/SIE.Web.Fixtures/Repairs/Commands/AddFixtureRepairCommand.js
SIE.defineCommand('SIE.Web.Fixtures.Repairs.Commands.AddFixtureRepairCommand', {
    extend: 'SIE.cmd.Add',
    meta: { text: "添加", group: "edit", iconCls: "iconfont icon-AddEntity icon-green" },
    /**
    * @override 执行
    * @param {} view 视图
    * @param {} source 
    * @returns {} 
    */
    execute: function (view, source) {
        var me = this;
        var editEntity = this.getEditEntity();
        view.execute({
            data: { Id: 0 },
            success: function (res) {
                if (res.Success) {
                    if (res.Result.errMsg) {
                        SIE.Msg.showWarning(res.Result.errMsg);
                    }
                    else {
                        res.Result.Id = editEntity.data.Id;
                        CRT.Workbench.addPage({
                            entityType: me.view.model,
                            recordId: editEntity.data.Id,
                            title: me.getEditViewTitle(editEntity),
                            isDetail: true,
                            params: {
                                No: res.Result.data.No,
                                RepairState: res.Result.data.RepairState,
                                ApplyById: res.Result.data.ApplyById,
                                ApplyDate: res.Result.data.ApplyDate,
                                ApplyByName: res.Result.data.ApplyByName,
                            }
                        });
                    }
                }
            }
        });
    }
});